hw/ip/dma/data/dma.hjson

```diff
@@ -0,0 +1,46 @@
+// Copyright EPFL contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+{ name: "dma",
+  clock_primary: "clk_i",
+  bus_interfaces: [
+    { protocol: "reg_iface", direction: "device" }
+  ],
+  regwidth: "32",
+  registers: [
+    { name:     "PTR_IN",
+      desc:     "Input data pointer",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "31:0", name: "PTR_IN", desc: "Input data pointer" }
+      ]
+    },
+    { name:     "PTR_OUT",
+      desc:     "Output data pointer",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        { bits: "31:0", name: "PTR_OUT", desc: "Output data pointer" }
+      ]
+    },
+    { name:     "DMA_START",
+      desc:     "Number of bytes to copy - Once a value is written, the copy starts",
+      swaccess: "rw",
+      hwaccess: "hrw",
+      fields: [
+        { bits: "31:0", name: "DMA_START", desc: "DMA counter and start" }
```

Comment:
although more efficient like you did, does it make sense to split start and counter in 2 different registers? I am thinking like you set the counter once, and the you start and stop later, not necessarily at the same time, not saying it's not ok, just reasoning

Reply:
Yes, I had exactly the same thinking but didn't find a strong advantage for one or the other. I decided to go for this one as you have to write to start the transfer anyway (and this way you can decide to change the number of bytes copied every time if needed). But maybe one solution will become better once the interface with the SPI is implemented.